### PR TITLE
Platformer - part 2

### PR DESCRIPTION
**Pull request by JVeg199X**

### CODE REVIEW

File: emu/src/emu_display.c [hunk 1]
```diff
@@ -642,9 +642,9 @@ void setLeds(led_t* leds, uint8_t numLeds)
 	pthread_mutex_lock(&displayMutex);
     for(int i = 0; i < numLeds; i++)
     {
-        rdLeds[i].r = leds[i].r >> ledBrightness;
-        rdLeds[i].g = leds[i].g >> ledBrightness;
-        rdLeds[i].b = leds[i].b >> ledBrightness;
+        rdLeds[i].r = leds[i].r;// >> ledBrightness;
```
Comment (AEFeinstein) [inline emu/src/emu_display.c:645]: Why did you remove emulated LED brightness control? If you set it to max brightness, it'd be equivalent. Please revert this/

File: main/gameData.c [hunk 2]
```diff
@@ -4,10 +4,7 @@
```
Comment (AEFeinstein) [inline main/gameData.c:4]: Please move this file into the `platformer` folder

File: main/gameData.c [hunk 3]
```diff
@@ -22,4 +19,58 @@
...
+    if(entityManager->playerEntity == NULL){
+        return;
+    }
+
+    uint8_t hp = entityManager->playerEntity->hp;
+    if(hp > 3){
+        hp = 3;
+    }
+
+    //HP meter led pairs:
```
Comment (AEFeinstein) [inline main/gameData.c:36]: There are 8 LEDs, not 6. I'm not sure if this was intentional or not, but you could use `NUM_LEDS` for loops

Comment (AEFeinstein) [reply main/gameData.c:36]: Nevermind, I see you use it later.

File: main/modes/platformer/entityManager.c [hunk 4]
```diff
@@ -63,6 +63,9 @@ void loadSprites(entityManager_t * entityManager)
     loadWsg("sprite021.wsg", &entityManager->sprites[SP_WARP_1]);
     loadWsg("sprite022.wsg", &entityManager->sprites[SP_WARP_2]);
     loadWsg("sprite023.wsg", &entityManager->sprites[SP_WARP_3]);
+    loadWsg("sprite024.wsg", &entityManager->sprites[SP_WASP_1]);
```
Comment (AEFeinstein) [inline main/modes/platformer/entityManager.c:66]: I'm not gonna require a change because you've got so many sprites, but next time try to use more descriptive and compact names. SPIFFS doesn't have folders for the output, so there's a chance of colliding with another mode's generically named sprites. I'd have done something like `ps_wasp1.wsg` for this line (`ps` for 'platformer sprite').

File: main/modes/platformer/gameData.h [hunk 5]
```diff
@@ -6,6 +6,13 @@
 //==============================================================================
 
 #include <stdint.h>
+#include "led_util.h"
+#include "common_typedef.h"
+
+//==============================================================================
+// Constants
+//==============================================================================
+#define NUM_LEDS 8 //Also defined in swadgeMode.h
```
Comment (AEFeinstein) [inline main/modes/platformer/gameData.h:15]: Please don't redefine this, just `#include "swadgeMode.h"` here. It's gonna stay as 8 this year, but I don't know for the future, and there's now precedence for porting modes from older swadges. Keep it portable!

File: main/modes/platformer/mode_platformer.c [hunk 6]
```diff
@@ -307,51 +308,19 @@ void updateTitleScreen(platformer_t *self)
 
     // Handle inputs
     if (
-        ((self->gameData.btnState & BTN_B) && !(self->gameData.prevBtnState & BTN_B)) 
-        ||
-        ((self->gameData.btnState & BTN_A) && !(self->gameData.prevBtnState & BTN_B))
```
Comment (AEFeinstein) [inline main/modes/platformer/mode_platformer.c:312]: I guess this had been a typo, but it's gone now!

File: main/modes/platformer/mode_platformer.c [hunk 7]
```diff
@@ -367,7 +336,7 @@ void drawPlatformerTitleScreen(display_t *d, font_t *font, gameData_t *gameData)
 
     if (gameData->frameCount < 10)
     {
-        drawText(d, font, c555, "Press A or B to start", 20, 128);
+        drawText(d, font, c555, "- Press START button -", 20, 128);
```
Comment (AEFeinstein) [inline main/modes/platformer/mode_platformer.c:339]: If you want to center text, don't use fixed offsets. Use `textWidth()` to find the width before drawing, then find the correct offset. For example:
```C
// Center and draw the text
int16_t tWidth = textWidth(&credits->radiostars, creditNames[idx]);
drawText(credits->disp, &credits->radiostars, creditColors[idx], creditNames[idx],
            (credits->disp->w - tWidth) / 2, (yPos + credits->yOffset));
```

File: main/modes/platformer/mode_platformer.c [hunk 8]
```diff
@@ -566,8 +541,8 @@ void updateGameClear(platformer_t *self){
 
 void drawGameClear(display_t *d, font_t *font, gameData_t *gameData){
     drawPlatformerHud(d, font, gameData);
-    drawText(d, font, c555, "Thanks for playing.", 24, 32);
-    drawText(d, font, c555, "Many more battle scenes", 8, 80);
-    drawText(d, font, c555, "will soon be available!", 8, 96);
-    drawText(d, font, c555, "Bonus 100000pts per life!", 8, 144);
+    drawText(d, font, c555, "Thanks for playing.", 24, 48);
```
Comment (AEFeinstein) [inline main/modes/platformer/mode_platformer.c:544]: Same note about centering text. You can also use a `yIdx` variable draw the text neatly, so start it at `48` and add `font->h + 4` after each line is drawn. That way you could swap fonts and keep a nice layout.

File: sdkconfig [hunk 9]
```diff
@@ -322,6 +322,7 @@ CONFIG_ESP32S2_TRACEMEM_RESERVE_DRAM=0x0
 # CONFIG_ESP32S2_ULP_COPROC_ENABLED is not set
```
Comment (AEFeinstein) [inline sdkconfig:322]: Can you revert his file please? It shouldn't have changed, but I guess the IDF tweaked it during a cmake build or something

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

File: main/modes/platformer/entityManager.c (hunk 4 above)
Comment (JVeg199X) [reply main/modes/platformer/entityManager.c:66]: They are named like this so that I can export all my sprites from my pixel art editor with 1 command, and also to keep the names consistent with where they are loaded into the sprites array.

File: emu/src/emu_display.c (hunk 1 above)
Comment (JVeg199X) [reply emu/src/emu_display.c]: Fixed.
I had set it this way because the LEDs were barely visible in my emulator.

File: main/gameData.c (hunk 2 above)
Comment (JVeg199X) [reply main/gameData.c]: Fixed

File: main/modes/platformer/gameData.h (hunk 5 above)
Comment (JVeg199X) [reply main/modes/platformer/gameData.h]: Fixed

File: main/modes/platformer/mode_platformer.c (hunk 6 above)
Comment (JVeg199X) [reply main/modes/platformer/mode_platformer.c:312]: It definitely was.

File: main/modes/platformer/mode_platformer.c (hunk 7 above)
Comment (JVeg199X) [reply main/modes/platformer/mode_platformer.c:339]: I'll keep this in mind. Will probably fix it later.

File: sdkconfig (hunk 9 above)
Comment (JVeg199X) [reply sdkconfig]: Fixed

File: main/modes/platformer/entityManager.c (hunk 4 above)
Comment (AEFeinstein) [reply main/modes/platformer/entityManager.c:66]: Gotcha

File: main/modes/platformer/mode_platformer.c (hunk 7 above)
Comment (AEFeinstein) [reply main/modes/platformer/mode_platformer.c:339]: As long as you're aware

**Review by AEFeinstein — APPROVED** [review]
